### PR TITLE
HLM-670 | API - Generate new Keycloak token

### PR DESCRIPTION
**Pull request by kuldeep-beehyv**
Create API to generate a token using user-id. @prasanna-egov  @saiprakash-v

### CODE REVIEW

File: backend/vc-management-service/src/controllers/tenant.controller.js [hunk 1]
```diff
@@ -56,6 +56,36 @@ async function createAndAssignNewRole(userName, token) {
     }
 }
 
+async function generateToken(req, res) {
+    const userId = req.params.userId.toLowerCase();
+    try {
+        const isValidUserId = utils.isValidUserId(userId);
+        if (isValidUserId) {
+            let token;
+            await keycloakService.getAdminToken().then(async (res) => {
```
Comment (prasanna-egov) [inline backend/vc-management-service/src/controllers/tenant.controller.js:65]: move this to within Keycloak service

File: backend/vc-management-service/src/services/keycloak.service.js [hunk 2]
```diff
@@ -91,18 +91,36 @@ const getRoleInfo = async (roleName, token) => {
...
     params.append('client_id', constants.SUNBIRD_SSO_CLIENT);
     params.append('client_secret', constants.SUNBIRD_SSO_ADMIN_CLIENT_SECRET);
 
+    const newtoken = await getToken(params);
+    return newtoken.access_token;
+}
+
+const generateUserToken = async (token, userId) => {
+    const params = new URLSearchParams();
+    params.append('grant_type', 'urn:ietf:params:oauth:grant-type:token-exchange');
```
Comment (prasanna-egov) [inline backend/vc-management-service/src/services/keycloak.service.js:106]: move to constants please

File: keycloak/Dockerfile [hunk 3]
```diff
@@ -3,4 +3,5 @@ FROM jboss/keycloak:16.1.1
 ADD themes /opt/jboss/keycloak/themes
 ADD providers /opt/jboss/keycloak/providers
 
-ENTRYPOINT ["/opt/jboss/tools/docker-entrypoint.sh"]
\ No newline at end of file
+ENTRYPOINT ["/opt/jboss/tools/docker-entrypoint.sh", "-Dkeycloak.profile.feature.token_exchange=enabled"]
```
Comment (prasanna-egov) [inline keycloak/Dockerfile:6]: Can we set this through environment variable. I think there is a way to enable experimental feature Token Exchange

**Comment by kuldeep-beehyv** [comment]
@prasanna-egov, and @saiprakash-v, incorporated all 3 comments.

**Comment by prasanna-egov** [comment]
WHat about changes to Keycloak. You also have  to make some changes to the Realm right. Please add the realm config file as well. @kuldeep-beehyv

**Review by prasanna-egov — APPROVED** [review]

**Pull request merged**